### PR TITLE
Dev jshint

### PR DESCRIPTION
**Pull request by itavy**
here are the changes to comply with the rules for jshint, jscs and eslint;
there are some changes needed to be done in the code but i suppresed the warnings there and commented there:

lib/server.js
 line 11 .. 16
 
lib/client.js
  line 518..525
  line 570..574
  line 657..679
  
lib/connect/tls.js
  line 9..13
  
lib/connect/tcp.js
  line 4..7
  
lib/connect/index.js
  line 76..81

next commit shall be to correct those coding style and remove the suppress messages

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+  events = require('events'),
+  Store = require('./store'),
+  eos = require('end-of-stream'),
+  mqttPacket = require('mqtt-packet'),
+  Writable = require('readable-stream').Writable,
+  inherits = require('inherits'),
+  setImmediate = global.setImmediate || function (callback) {
+    // works in node v0.8
+    process.nextTick(callback);
+  };
```
Comment (mcollina) [inline lib/client.js]: I actually prefer to have module dependencies at the top, and then all the rest.

File: lib/client.js [hunk 2]
```diff
@@ -23,11 +26,11 @@ var defaultConnectOptions = {
   clean: true
 };
```
Comment (mcollina) [inline lib/client.js]: you can define this in the top block

File: lib/client.js [hunk 3]
```diff
@@ -23,11 +26,11 @@ var defaultConnectOptions = {
...
 
-var defaultId = function() {
+defaultId = function () {
   return 'mqttjs_' + Math.random().toString(16).substr(2, 8);
 };
```
Comment (mcollina) [inline lib/client.js]: you can define it as a function, `function defaultId`, so it follows the same pattern of `nop` behind.

File: lib/client.js [hunk 4]
```diff
@@ -36,9 +39,9 @@ function nop() {}
  * @param {Object} [options] - connection options
  * (see Connection#connect)
  */
-var MqttClient = module.exports =
-function MqttClient(streamBuilder, options) {
-  var that = this;
+MqttClient = module.exports = function MqttClient (streamBuilder, options) {
```
Comment (mcollina) [inline lib/client.js]: This line is too long. Move `module.expors` at the bottom and remove the whole `var` thing.

File: lib/client.js [hunk 5]
```diff
@@ -504,14 +511,22 @@ MqttClient.prototype._cleanUp = function(forced, done) {
...
 
-function sendPacket(client, packet, cb) {
+/*
+ * to comply with strict rules, a function must be
+ * declared before it can be used
+ * so i moved it has to be  moved before its first call
+ * later on maybe we can move all of them to the top of the file
+ * for now i just suppressed the warning
+ */
+/*jshint latedef:false*/
```
Comment (mcollina) [inline lib/client.js]: Move it at the top, no problem for me.

File: lib/connect/tls.js [hunk 6]
```diff
@@ -1,26 +1,42 @@
...
+  var connection;
+  opts.port = opts.port || 8883;
+  opts.host = opts.hostname || opts.host || 'localhost';
+
+  /**
+   * this should be further investigated
+   * the result is opts.rejectUnauthorized = opts.rejectUnauthorized
+   * do you want to check for undefined and set it to false default ?
+   */
+  /*jshint ignore:start */
```
Comment (mcollina) [inline lib/connect/tls.js]: Can't remember, feel free to change/improve.

File: lib/server.js [hunk 7]
```diff
@@ -1,26 +1,35 @@
...
-  , util = require('util')
-  , Connection = require('mqtt-connection');
-
-function setupConnection(duplex) {
+/*
+ * jshint comply here since using "this" in
+ * a function that is not likely to be a method
+ * breaks the strict rules
+ * a solution will be to add the caller as parameter to it
+ */
```
Comment (mcollina) [inline lib/server.js]: Remove the comment and keep ignoring it. It is meant to be used this way.

**Comment by mcollina** [comment]
Thanks very much for the great work put into this. Anybody else would like to review the changes before we merge in? Thanks!

**Comment by itavy** [comment]
i have made all the required changes and i have moved required on top before uninitialized variables on all the files.

**Comment by mcollina** [comment]
@adamvr @256dpi @mdickens what do you think? For me is LGTM.

**Comment by 256dpi** [comment]
looks cool! :+1:

**Comment by mcollina** [comment]
I will merge this on Monday if nobody else has objections.

**Comment by mcollina** [comment]
@itavy thanks so much! I added you to this organization! Feel free to add yourself to the README.md.

**Pull request merged**